File: 0264-ugly-number-ii/0264-ugly-number-ii.py
class Solution:
    def nthUglyNumber(self, n: int) -> int:
        nums = [1]
        i2 = i3 = i5 = 0
        while len(nums) < n:
            t2, t3, t5 = nums[i2] * 2, nums[i3] * 3, nums[i5] * 5
            nums.append(min(t2, t3, t5))
            if nums[-1] == t2:
                i2 += 1
            if nums[-1] == t3:
                i3 += 1
            if nums[-1] == t5:
                i5 += 1
        return nums[-1]

File: 0264-ugly-number-ii/0264-ugly-number-ii.js
/**
 * @param {number} n
 * @return {number}
 */
var nthUglyNumber = function(n) {
  const nums = [1];
  let [i, i2, i3, i5] = [1, 0, 0, 0];
  
  while (i < n) {
    const [t2, t3, t5] = [nums[i2] * 2, nums[i3] * 3, nums[i5] * 5];
    nums[i] = Math.min(t2, t3, t5);
    
    if (nums[i] === t2) i2++;
    if (nums[i] === t3) i3++;
    if (nums[i] === t5) i5++;
    i++;
  }
  
  return nums[n - 1];
};